tests/unittest_nnstreamer-edge.cc

```diff
@@ -2815,6 +2815,460 @@ TEST(edgeMeta, deserializeInvalidParam03_n)
+        EXPECT_EQ (ret, NNS_EDGE_ERROR_NONE);
+
+        EXPECT_EQ (count, 1U);
+        for (i = 0; i < 10U; i++)
+          EXPECT_EQ (((unsigned int *) data)[i], i);
+      }
+
+      ret = nns_edge_data_destroy (data_h);
+      EXPECT_EQ (ret, NNS_EDGE_ERROR_NONE);
+      break;
+    default:
+      break;
+  }
+
+  return NNS_EDGE_ERROR_NONE;
+}
+
+/**
+ * @brief Check whether MQTT broker is running or not.
+ */
+static int
+_check_mqtt_broker ()
+{
+  char *check_broker_cmd;
+  int ret = 0;
+
+  check_broker_cmd = nns_edge_strdup ("ps aux | grep mosquitto | grep -v grep");
+  ret = system (check_broker_cmd);
+  if (0 != ret) {
+    nns_edge_loge ("MQTT broker is not running. Skip query hybrid test.");
+    ret = -1;
+  }
+
+  return ret;
+}
+
+/**
+ * @brief Connect to the local host using the information received from mqtt.
+ */
+TEST(edgeMqtt, connectLocal)
+{
+  nns_edge_h server_h, client_h;
+  ne_test_data_s *_td_server, *_td_client;
+  nns_edge_data_h data_h;
+  pthread_t server_thread, client_thread;
+  pthread_attr_t attr;
+  size_t data_len;
+  void *data;
+  unsigned int i, retry;
+  int ret = 0;
+  char *val;
+
+  if (0 != _check_mqtt_broker ())
+    return;
+
+  _td_server = _get_test_data (true);
+  _td_client = _get_test_data (false);
+
+  /* Prepare server (127.0.0.1:port) */
+  nns_edge_create_handle ("temp-server", NNS_EDGE_CONNECT_TYPE_HYBRID, NNS_EDGE_FLAG_RECV | NNS_EDGE_FLAG_SEND | NNS_EDGE_FLAG_SERVER, &server_h);
+  nns_edge_set_event_callback (server_h, _test_edge_hybrid_event_cb, _td_server);
+  nns_edge_set_info (server_h, "HOST", "localhost");
+  nns_edge_set_info (server_h, "PORT", "0");
+  nns_edge_set_info (server_h, "DEST_HOST", "tcp://localhost");
+  nns_edge_set_info (server_h, "DEST_PORT", "1883");
+  nns_edge_set_info (server_h, "TOPIC", "temp-mqtt-topic");
+  nns_edge_set_info (server_h, "CAPS", "test server");
+  _td_server->handle = server_h;
+
+  /* Prepare client */
+  nns_edge_create_handle ("temp-client", NNS_EDGE_CONNECT_TYPE_HYBRID, NNS_EDGE_FLAG_RECV | NNS_EDGE_FLAG_SEND, &client_h);
+  nns_edge_set_event_callback (client_h, _test_edge_hybrid_event_cb, _td_client);
+  nns_edge_set_info (client_h, "CAPS", "test client");
+  nns_edge_set_info (client_h, "HOST", "localhost");
+  nns_edge_set_info (client_h, "port", "0");
+  nns_edge_set_info (client_h, "TOPIC", "temp-mqtt-topic");
+  _td_client->handle = client_h;
+
+  /* Start server/client thread */
+  pthread_attr_init (&attr);
+  pthread_attr_setdetachstate (&attr, PTHREAD_CREATE_DETACHED);
+  pthread_create (&server_thread, &attr, _test_edge_thread, _td_server);
+  pthread_create (&client_thread, &attr, _test_edge_thread, _td_client);
+  pthread_attr_destroy (&attr);
+
+  /* Wait for server/client thread */
+  do {
+    usleep (20000);
+  } while (!g_main_loop_is_running (_td_server->loop));
+
+  do {
+    usleep (20000);
+  } while (!g_main_loop_is_running (_td_client->loop));
+
+  ret = nns_edge_connect (client_h, "tcp://localhost", 1883);

```

Comment:
line 2930 and 2960, how about getting available port using nns_edge_get_available_port()?

Reply:
This is for mosquitto broker port. Need to set 1883.